libs/effects/Effect/System.idr

```diff
@@ -14,18 +15,23 @@ data System : Effect where
      Time : sig System Integer
      GetEnv : String -> sig System (Maybe String)
      CSystem : String -> sig System Int
+     Usleep : (i : Int) ->
+              {auto prf : So (i >= fromInteger 0 && Delay (i <= fromInteger 1000000))} ->
```

Comment:
Any reason you use `fromInteger` here explicitly? literals should be overloaded. The same with `Delay`.